File: Amit-Kumar-Singh/Two_Pointer/Reverse_Only_Letters.java
class Solution {
    public String reverseOnlyLetters(String s) {
        char[] str = s.toCharArray();
        int start = 0;
        int end = s.length()-1;
        while(start<end){
            if((str[start]>='a' && str[start]<='z') || (str[start]>='A' && str[start]<='Z')){
                if((str[end]>='a' && str[end]<='z') || (str[end]>='A' && str[end]<='Z')){
                    char temp = str[start];
                    str[start] = str[end];
                    str[end] = temp;
                    start++;
                    end--;
                }
                else{
                    end--;
                }
            }
            else{
                start++;
            }
        }
        return new String(str);
    }
}

File: Lakshay Garg/Leetcode/Two_Pointers_Leetcode/Reverse_Only_Letters.cpp
class Solution {
public:
    string reverseOnlyLetters(string s) {
        int i=0,j=s.size()-1;
        while(i<j){
            if(!isalpha(s[i])){
                i++;
                continue;
            }
            if(!isalpha(s[j])){
                j--;
                continue;
            }
            swap(s[i],s[j]);
            i++;
            j--;
        }
        return s;
    }
};
